### PR TITLE
Resolve PodSecurityAdmission restrictions on 1.23+ for deprecated PodSecurityPolicy

### LINKED ISSUE #5603: Container configurations need to be updated for `PodSecurity` on k8s 1.23+
I've verified this on 1.23 and 1.24 in clusters created via kind. The error message is the same.

The deployments fail to create pods - from `kubectl get deployment -n tekton-pipelines tekton-pipelines-controller`:

```
    message: 'pods "tekton-pipelines-controller-fcfbc554b-9fcdj" is forbidden: violates
      PodSecurity "restricted:latest": seLinuxOptions (pod set forbidden securityContext.seLinuxOptions:
      ), unrestricted capabilities (container "tekton-pipelines-controller" must set
      securityContext.capabilities.drop=["ALL"]), seccompProfile (pod or container
      "tekton-pipelines-controller" must set securityContext.seccompProfile.type to
      "RuntimeDefault" or "Localhost")'
```

This is due to https://github.com/tektoncd/pipeline/commit/b506b776da7dad3c30d65decfb405540320d4fe9#diff-d98ea73731da6e49a8552d6917d796aab6613198b269eda1b4bbfb4b53d90111R22, added in #5536 as part of addressing #4112.

While the move to `PodSecurity` is being reverted in #5605, this issue will be used to track the additional container configuration work needed to move to `PodSecurity`.

### PR DESCRIPTION
**Pull request by JeromeJu**
# Changes
This commit fixes the issue where the securityContext are not restricted in PodSecurityAdmission(PSA). This removes the PodSeucrityPolicy, which is deprecated in Kubernetes v1.21 and removed from v1.25. This adds to the PSA restricted label with respective policies enforced by PSP but not covered by the restricted standard of PSA.

The previous settings in 101-podsecuritypolicy.yaml are restricting roles to be run as nonRoot, this is covered by the PodSecurityStandard of the restricted level.

/kind bug
Fixes #5603 #4112 
# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [n/a] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [n/a] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [n/a] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [n/a] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
Action required: If using Kubernetes 1.22, set PodSecurity flag to true to enforce a restricted pod security level in Tekton namespaces. See https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/#feature-gates-for-graduated-or-deprecated-features for more information.
```

### CODE REVIEW

**Comment by JeromeJu** [comment]
cc @abayer I've tested against [1.23.12 and 1.24.6](https://github.com/kubernetes-sigs/kind/releases) cluster locally and didn't encounter the error after `kubectl get deployment -n tekton-pipelines tekton-pipelines-controller -o yaml`. Could you please help test against the original context of #5603?

File: test.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+kind: Cluster
```
Comment (abayer) [inline test.yaml]: I'm guessing this wasn't supposed to be added. =)

**Comment by abayer** [comment]
@JeromeJu I'm still getting the same error - I used this kind config YAML:
```
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
name: kind1.23.12
nodes:
- role: control-plane
  image: kindest/node:v1.23.12@sha256:9402cf1330bbd3a0d097d2033fa489b2abe40d479cc5ef47d0b6a6960613148a
- role: worker
  image: kindest/node:v1.23.12@sha256:9402cf1330bbd3a0d097d2033fa489b2abe40d479cc5ef47d0b6a6960613148a
```
..and ran `ko apply -R -f config`. Then `kc get deployment -n tekton-pipelines tekton-pipelines-controller -o yaml` again, and got the same thing:
```
status:
  conditions:
  - lastTransitionTime: "2022-10-18T14:47:31Z"
    lastUpdateTime: "2022-10-18T14:47:31Z"
    message: Created new replica set "tekton-pipelines-controller-6854456f96"
    reason: NewReplicaSetCreated
    status: "True"
    type: Progressing
  - lastTransitionTime: "2022-10-18T14:47:31Z"
    lastUpdateTime: "2022-10-18T14:47:31Z"
    message: Deployment does not have minimum availability.
    reason: MinimumReplicasUnavailable
    status: "False"
    type: Available
  - lastTransitionTime: "2022-10-18T14:47:31Z"
    lastUpdateTime: "2022-10-18T14:47:31Z"
    message: 'pods "tekton-pipelines-controller-6854456f96-vbxdx" is forbidden: violates
      PodSecurity "restricted:latest": unrestricted capabilities (container "tekton-pipelines-controller"
      must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod
      or container "tekton-pipelines-controller" must set securityContext.runAsNonRoot=true),
      seccompProfile (pod or container "tekton-pipelines-controller" must set securityContext.seccompProfile.type
      to "RuntimeDefault" or "Localhost")'
    reason: FailedCreate
    status: "True"
    type: ReplicaFailure
  observedGeneration: 1
  unavailableReplicas: 1
```

File: test.yaml (hunk 1 above)
Comment (JeromeJu) [reply test.yaml]: Yes sorry, I updated this in a later commit.

**Comment by lbernick** [comment]
@abayer I know our CI runs e2e tests on kind w/ k8s version 1.23 now and I can see you've tested on 1.24 as well. Do we have a way of testing this on all the k8s versions we support via CI, or do you think it's OK to merge this?

**Comment by abayer** [comment]
There's no easy way to test them all in CI, sadly, but yeah, this is good to merge, IMO.

**Pull request merged**